code/modules/antagonists/changeling/powers/mutations.dm

```diff
@@ -18,28 +18,40 @@
 	var/weapon_name_simple
 
 /datum/action/changeling/weapon/try_to_sting(mob/user, mob/target)
-	if(istype(user.l_hand, weapon_type)) //Not the nicest way to do it, but eh
-		qdel(user.l_hand)
-		if(!silent)
-			user.visible_message("<span class='warning'>With a sickening crunch, [user] reforms [user.p_their()] [weapon_name_simple] into an arm!</span>", "<span class='notice'>We assimilate the [weapon_name_simple] back into our body.</span>", "<span class='warning'>You hear organic matter ripping and tearing!</span>")
-		user.update_inv_l_hand()
-		return
-	if(istype(user.r_hand, weapon_type))
-		qdel(user.r_hand)
-		if(!silent)
-			user.visible_message("<span class='warning'>With a sickening crunch, [user] reforms [user.p_their()] [weapon_name_simple] into an arm!</span>", "<span class='notice'>We assimilate the [weapon_name_simple] back into our body.</span>", "<span class='warning'>You hear organic matter ripping and tearing!</span>")
-		user.update_inv_r_hand()
+	if(istype(user.l_hand, weapon_type) || istype(user.r_hand, weapon_type))
+		retract(user, TRUE)
 		return
 	..(user, target)
 
 /datum/action/changeling/weapon/sting_action(mob/user)
+	SEND_SIGNAL(user, COMSIG_MOB_WEAPON_APPEARS)
 	if(!user.drop_item())
 		to_chat(user, "[user.get_active_hand()] is stuck to your hand, you cannot grow a [weapon_name_simple] over it!")
 		return FALSE
-	var/obj/item/W = new weapon_type(user, silent)
+	var/obj/item/W = new weapon_type(user, silent, src)
 	user.put_in_hands(W)
+	RegisterSignal(user, COMSIG_MOB_WILLINGLY_DROP, PROC_REF(retract))
+	RegisterSignal(user, COMSIG_MOB_WEAPON_APPEARS, PROC_REF(retract))
```

Comment:
To denote these are meant to be overriden in this proc.
```suggestion
	RegisterSignal(user, COMSIG_MOB_WILLINGLY_DROP, PROC_REF(retract), override = TRUE)
	RegisterSignal(user, COMSIG_MOB_WEAPON_APPEARS, PROC_REF(retract), override = TRUE)
```

Reply:
updated https://github.com/ParadiseSS13/Paradise/pull/21080/commits/f4678bed9311c276c71dc5ad91932dbef567f4e5
though, those signals would never make runtime with overriding